services/QuillLMS/engines/evidence/spec/controllers/evidence/feedback_controller_spec.rb

```diff
@@ -23,6 +24,287 @@ module Evidence
+
+        it 'should return successfully when there is regex3 feedback' do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::RegexTypo])
+          post :create, params: {entry: "test regex response", prompt_id: prompt.id, session_id: 1, previous_feedback: ([]) }, as: :json
+
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+        end
+      end
+
+      context 'plagiarism test' do
+        it 'should return successfully when there is plagiarism that matches the second plagiarism text string' do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::Plagiarism])
+          post :create, params: { entry: "bla bla bla #{plagiarized_text2}", prompt_id: prompt.id, session_id: 1, previous_feedback: [] }, as: :json
+
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+          expect(plagiarized_text2).to(eq(parsed_response["highlight"][0]["text"]))
+          expect(plagiarized_text2).to(eq(parsed_response["highlight"][1]["text"]))
+          expect(first_feedback.text).to(eq(parsed_response["feedback"]))
+          request.env.delete("RAW_POST_DATA")
+          post("plagiarism", :params => ({ :entry => ("bla bla bla #{plagiarized_text2}"), :prompt_id => prompt.id, :session_id => 5, :previous_feedback => ([parsed_response]) }), :as => :json)
+          parsed_response = JSON.parse(response.body)
+          expect(parsed_response["optimal"]).to be false
+          expect(second_feedback.text).to(eq(parsed_response["feedback"]))
+        end
+      end
+
+      context 'prefilter test' do
+        let!(:profanity_rule) do
+          create(:evidence_rule, rule_type: 'prefilter', uid: 'fdee458a-f017-4f9a-a7d4-a72d1143abeb')
+        end
+
+        before do
+          stub_const("Evidence::Check::ALL_CHECKS", [Check::Prefilter])
+        end
+
+        context 'profanity violation' do
+          it 'should return correct rule uid' do
+            post :create, params: {
+              entry: "nero was an ahole",
+              prompt_id: prompt.id,
+              session_id: 1,
+              previous_feedback: ([])
+            }, :as => :json
+            expect(response.status).to eq 200
+            parsed_response = JSON.parse(response.body)
+            expect(parsed_response['optimal']).to be false
+            expect(parsed_response['rule_uid']).to eq profanity_rule.uid
+          end
+        end
+
+        context 'no violation' do
+          it 'should return empty response' do
+            post :create, params: {
+              entry: "the quick brown fox",
+              prompt_id: prompt.id,
+              session_id: 1,
+              previous_feedback: ([])
+            }, :as => :json
+
+            expect(response.status).to eq 200
+            parsed_response = JSON.parse(response.body)
+            expect(parsed_response).to eq({})
```

Comment:
This is not a comment on this particular test, but I think this is the only case we test that confirms that the controller returns `{}` even if all the checks error or fail somehow.  Might be worth having an explicit case for the "fallback" case rather than just having an incidental test for that case inside the "prefilter" context.

Reply:
Good point, I'll just remove this test and add one for the general case.